### PR TITLE
Refactored IndexerTest to use conditional test execution annotation

### PR DESCRIPTION
**Pull request by eas5**
Problem:
Conditions within the test method may produce a false test outcome. Once a condition is not met, and the code block inside the condition is not executed, a false 'passed' outcome can be produced without the test ever being executed.

Solution:
Instead of using a conditional to control the test execution, we propose using the Conditional Test Execution Annotation @EnabledIf, which results in a test being aborted before a previous test. Conditional execution is typically used whenever it does not make sense to continue executing a given test method.

### CODE REVIEW

File: opengrok-indexer/src/test/java/org/opengrok/indexer/index/IndexerTest.java [hunk 1]
```diff
@@ -451,39 +452,32 @@ public void testIncrementalIndexAddRemoveFile() throws Exception {
...
+        idb.addIndexChangedListener(listener);
+        System.out.println("Trying to index a special file - FIFO in this case.");
+        idb.update();
+        assertEquals(0, listener.files.size());
+    }
 
-        } else {
-            System.out.println("Skipping test for bug 11896. Could not find a mkfifo in path.");
-        }
+    public boolean mkinfoInPath() {
```
Comment (ahornace) [inline opengrok-indexer/src/test/java/org/opengrok/indexer/index/IndexerTest.java:479]: This should be named `mkfifoInPath`

Comment (ahornace) [inline opengrok-indexer/src/test/java/org/opengrok/indexer/index/IndexerTest.java:479]: No need for `public` modifier, can be even `private` it seems

**Review by ahornace — CHANGES_REQUESTED** [review]

File: opengrok-indexer/src/test/java/org/opengrok/indexer/index/IndexerTest.java (hunk 1 above)
Comment (ahornace) [reply opengrok-indexer/src/test/java/org/opengrok/indexer/index/IndexerTest.java:479]: https://junit.org/junit5/docs/snapshot/user-guide/index.html#writing-tests-conditional-execution-custom has an example with package private access so I think we should go with that :)

**Comment by eas5** [comment]
Thank you for your revision @ahornace. I have made the suggested changes :)

**Comment by ahornace** [comment]
> Thank you for your revision @ahornace. I have made the suggested changes :)

Cool! Thanks for the PR :) I'll merge as soon as the checks pass

**Pull request merged**